### PR TITLE
feat!: initial generation of library

### PR DESCRIPTION
**Pull request by steffnay**

### CODE REVIEW

File: linkinator.config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "recurse": true,
+  "skip": [
+    "https://codecov.io/gh/googleapis/",
+    "www.googleapis.com",
+    "img.shields.io",
+    "https://www.npmjs.org/package/@google-cloud/bigquery-storage",
+    "https://github.com/googleapis/nodejs-bigquery-storage/tree/master/samples",
```
Comment (JustinBeckwith) [inline linkinator.config.json]: Please file a bug to fast-followup removing these

File: package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "@google-cloud/bigquery-storage",
+  "version": "0.1.0",
+  "description": "Storage client for Node.js",
```
Comment (JustinBeckwith) [inline package.json]: Could we say "Client for the BigQuery Storage API"?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  "repository": "googleapis/nodejs-storage",
```
Comment (JustinBeckwith) [inline package.json]: womp womp

File: package.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "docs-test": "linkinator docs",
+    "fix": "gts fix && eslint samples --fix",
+    "lint": "gts check && eslint samples",
+    "prepare": "npm run compile-protos && npm run compile",
+    "system-test": "c8 mocha build/system-test",
+    "test": "c8 mocha build/test",
+    "samples-test": "cd samples/ && npm link ../ && npm test && cd ../"
+  },
+  "dependencies": {
+    "google-gax": "^1.12.0"
```
Comment (JustinBeckwith) [inline package.json]: this should be 1.14.1

File: package.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  "devDependencies": {
+    "@types/mocha": "^5.2.5",
+    "@types/node": "^12.0.0",
+    "c8": "^6.0.0",
+    "gts": "^1.0.0",
+    "jsdoc": "^3.5.5",
+    "jsdoc-fresh": "^1.0.1",
+    "jsdoc-region-tag": "^1.0.2",
+    "linkinator": "^1.5.0",
```
Comment (JustinBeckwith) [inline package.json]: some of these are out of date.  Can you try doing a `npx npm-upgrade`?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    "webpack": "^4.41.2",
+    "webpack-cli": "^3.3.10",
+    "eslint": "^6.0.0",
+    "eslint-config-prettier": "^6.0.0",
+    "eslint-plugin-node": "^10.0.0",
+    "eslint-plugin-prettier": "^3.0.1",
+    "prettier": "^1.17.0"
+  },
+  "engines": {
+    "node": ">=8.13.0"
```
Comment (JustinBeckwith) [inline package.json]: @bcoe do we want to start this at 10?

File: .repo-metadata.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    "name_pretty": "Google BigQuery Storage",
+    "product_documentation": "https://cloud.google.com/bigquery/docs/reference/storage",
+    "issue_tracker": "https://b.corp.google.com/savedsearches/559654",
+    "release_level": "beta",
+    "language": "nodejs",
+    "repo": "googleapis/nodejs-bigquery-storage",
+    "distribution_name": "@google-cloud/bigquery-storage",
+    "api_id": "bigquerystorage.googleapis.com",
+    "requires_billing": true
+  }
```
Comment (JustinBeckwith) [inline .repo-metadata.json]: please add a new line at the end

File: samples/package.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  "dependencies": {
+    "@google-cloud/bigquery-storage": "^1.0.1",
+    "avsc": "^5.4.18",
+    "snappy": "^6.2.3"
+  },
+  "devDependencies": {
+    "c8": "^5.0.1",
+    "chai": "^4.2.0",
+    "mocha": "^6.2.2"
```
Comment (JustinBeckwith) [inline samples/package.json]: This should be at 7

File: samples/quickstart.js [hunk 9]
```diff
@@ -0,0 +1,143 @@
+/**
```
Comment (JustinBeckwith) [inline samples/quickstart.js]: The header here doesn't match headers elsewhere.  Uses /* */ style over // style

File: samples/quickstart.js [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+async function main() {
+  // [START bigquery_storage_quickstart]
+  async function bigqueryStorageQuickstart() {
+    // The read stream contains blocks of Avro-encoded bytes. We use the
```
Comment (JustinBeckwith) [inline samples/quickstart.js]: Lines 21 through 31 should go *outside* of the async function, and inside of `main`.

File: samples/quickstart.js [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
```
Comment (JustinBeckwith) [inline samples/quickstart.js]: Can we add a sample header here with title, description, and usage info?  For example:

```

// sample-metadata:
//   title: Export Assets
//   description: Export asserts to specified dump file path.
//   usage: node exportAssets.js <gs://my-bucket/my-assets.txt>
```

File: samples/package.json [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    "node": ">=8"
+  },
+  "files": [
+    "*.js"
+  ],
+  "scripts": {
+    "test": "c8 mocha --timeout 600000 test/*.js"
+  },
+  "dependencies": {
+    "@google-cloud/bigquery-storage": "^1.0.1",
```
Comment (bcoe) [inline samples/package.json]: I believe this should be `@google-cloud/bigquery-storage": "^0.1.0", to match the current value in the top level `package.json`.

File: package.json [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+  "main": "build/src/index.js",
+  "scripts": {
+    "clean": "gts clean",
+    "compile": "tsc -p . && cp -r protos build/",
+    "compile-protos": "compileProtos src",
+    "docs": "jsdoc -c .jsdoc.js",
+    "predocs-test": "npm run docs",
+    "docs-test": "linkinator docs",
+    "fix": "gts fix && eslint samples --fix",
+    "lint": "gts check && eslint samples",
```
Comment (bcoe) [inline package.json]: If you add the line:

```
    "prelint": "cd samples; npm link ../; npm i"
```

I think you will find things start working for you, as soon as you fix the version #.

**Pull request merged**